src/Cli/dotnet/OptionForwardingExtensions.cs

```diff
@@ -19,7 +19,11 @@ public static class OptionForwardingExtensions
         public static ForwardedOption<T> ForwardAsSingle<T>(this ForwardedOption<T> option, Func<T, string> format) => option.SetForwardingFunction(format);
 
         public static ForwardedOption<string[]> ForwardAsProperty(this ForwardedOption<string[]> option) => option
-            .SetForwardingFunction((optionVals) => optionVals.SelectMany(optionVal => new string[] { $"{option.Aliases.FirstOrDefault()}:{optionVal.Replace("roperty:", string.Empty)}" }));
+            .SetForwardingFunction((optionVals) =>
+                optionVals
+                    .Select(optionVal => optionVal.Replace(";", "%3B")) // must escape semicolon-delimited property values when forwarding them to MSBuild
+                    .SelectMany(optionVal => new string[] { $"{option.Aliases.FirstOrDefault()}:{optionVal.Replace("roperty:", string.Empty)}" })
```

Comment:
Will this work correctly if you do something like `-p:MyCrazyProperty=property:value`, or will it remove the `roperty:` portion of the property value?

Reply:
good catch - I'll add some tests around this forwarding function specifically (by extracting it to a named function and slamming it with different inputs), but IMO we should be canonicalizing all of the different input shapes into `-property:<user input>` for MSBuild (meaning that we use the most verbose form available).